File: lab05/src/hust/soict/itep/aims/media/Track.java
package hust.soict.itep.aims.media;

public class Track implements Playable{
    
    private String title;
    private int length;

    public Track(String title, int length) {
        this.title = title;
        this.length = length;
    }

    public String getTitle() {
        return title;
    }

    public int getLength() {
        return length;
    }

    // Trien khai phuong thuc play() trong interface
    @Override
    public void play() {
        System.out.println("HoTuanAnh-20226100 Playing track: " + this.getTitle());
        System.out.println("HoTuanAnh-20226100 Track length: " + this.getLength());
    }

    // Ghi de phuong thuc equals() mac dinh
    @Override
    public boolean equals(Object o) {
        Track track = (Track) o;
        return track.getTitle().equals(this.getTitle()) && (track.getLength() == this.getLength());
    }
}

File: lab04/src/hust/soict/itep/aims/media/Track.kt
package hust.soict.itep.aims.media

class Track(var title: String, var length: Int) : Playable {

    override fun play() {
        println("Playing track: $title")
        println("Track length: $length minutes")
    }

    override fun equals(other: Any?): Boolean {
        val track = other as Track
        return track.title == this.title && track.length == this.length
    }

    override fun hashCode(): Int {
        var result = length
        result = 31 * result + title.hashCode()
        return result
    }
}